packages/jss/src/index.d.ts

```diff
@@ -67,80 +191,47 @@ export interface StyleSheet<RuleName extends string = any> {
   /**
    * Update the function values with a new data.
    */
-  update(data?: {}): this;
-  update(name: RuleName, data: {}): this;
+  update(name: string, data: object, options?: UpdateOptions): this
+  update(data: object, options?: UpdateOptions): this
   /**
    * Convert rules to a CSS string.
    */
   toString(options?: ToCssOptions): string;
 }
-export type GenerateClassName<Name extends string = any> = (rule: Rule, sheet?: StyleSheet<Name>) => string;
-
-export interface JSSPlugin {
```

Comment:
It looks like you've changed both the order and the shape of some of these types.  That makes it really hard to review.

Reply:
Yeah, because the one was renamed and the other typing was wrong.